### PR TITLE
Make Compatible with Bazel 5.0

### PR DESCRIPTION
**Pull request by gnossen**
[A breaking change](https://github.com/bazelbuild/bazel/issues/14260) has been (unintentionally?) made in Bazel 5.0 wherein the `android_ndk_repository` rule is evaluated eagerly. That is, even when no Android targets are being evaluated. This forces users of a monorepo to have an Android SDK installed even if they are not working on Android.

This is an unacceptable burden. This PR checks for environment variables indicating the presence of an Android SDK/NDK installation and turns the Android dependency into an empty target if they are not found. This slightly complicates the Android dependency, but makes the development process reasonable once again.

Credit to @keith from the Envoy Mobile team for the idea.

There are three use cases to consider here:

### CI invocations within grpc/grpc

I've explicitly added the path to the Android SDK/NDK in a build script common to all of the Android Kokoro jobs. This ensures that those jobs that actually depend on Android will have access to the appropriate env vars.

### Manual Developer Invocations within the `grpc/grpc` repo

This workflow was broken, as explained above. Manual testing shows that the workflow is now fixed. There was a bit of trouble with Windows paths that I *believe* I've solved, but I do not have access to a Windows machine to test this more deeply.

### Developer invocations from repos depending on `grpc/grpc`

The `python_bazel_test_repo` test verifies that this was never broken because our Android repo dependency lives only in the `WORKSPACE` file, not in `grpc_deps` or `grpc_extra_deps`.

### CODE REVIEW

**Comment by gnossen** [comment]
Applying DO NOT MERGE label until I've removed the upgrade to Bazel 5.0.0. That part is just for testing things in CI.

**Comment by gnossen** [comment]
Tests are looking good. I'll remove the change to Bazel 5 and then this should be good to merge.

**Review by veblush — APPROVED** [review]

File: tools/internal_ci/helper_scripts/prepare_build_linux_rc [hunk 1]
```diff
@@ -18,6 +18,9 @@
 # Need to increase open files limit for c tests
 ulimit -n 32768
 
+# This is required by the android_ndk_repository repo rule.
+export ANDROID_NDK_HOME=$ANDROID_HOME/ndk-bundle
```
Comment (jtattermusch) [inline tools/internal_ci/helper_scripts/prepare_build_linux_rc:22]: Does this mean that unless the ANDROID_NDK_HOME variable is set, bazel 5 build is going to fail?
that seems pretty rought. 
- does the android NDK actually have to be present? or it's just that that env variable needs to be set. If it actually needs to be installed, that's even worse.
- do we really set this env variable for ALL our linux builds?

Also, it looks like this hack won't cover the situations where we run bazel build under docker (e.g. bazel distribtests).  We should probably add a distribtest for bazel 5 ASAP.

**Review by jtattermusch — COMMENTED** [review]
@gnossen  I had some concerns about the path taken - let's address as a followup.

File: tools/internal_ci/helper_scripts/prepare_build_linux_rc (hunk 1 above)
Comment (jtattermusch) [reply tools/internal_ci/helper_scripts/prepare_build_linux_rc:22]: Thanks for the explanation.
Let's move the export ANDROID_NDK_HOME out of `prepare_build_linux_rc` and only add it to the jobs that actually need to build android.

I think the affected jobs would be 
grpc/core/master/linux/grpc_binder_transport_apk

and perhaps also 
grpc/core/master/linux/grpc_android

**Pull request merged**